tests/next_tests/definitions.py

```diff
@@ -55,6 +59,32 @@ class ProgramBackendId(_PythonObjectIdMixin, str, enum.Enum):
     DOUBLE_ROUNDTRIP = "gt4py.next.program_processors.runners.double_roundtrip.backend"
 
 
+class ExecutionAndAllocatorDescriptor(Protocol):
+    # Used for test infrastructure, consider implementing this in gt4py when refactoring otf
+    @property
+    def executor(self) -> Optional[ppi.ProgramExecutor]:
+        ...
+
+    @property
+    def allocator(self) -> next_allocators.FieldBufferAllocatorProtocol:
+        ...
+
+
+@dataclasses.dataclass(frozen=True)
+class EmbeddedExecutionDescriptor:
+    allocator: next_allocators.FieldBufferAllocatorProtocol
+    executor: NoneType = dataclasses.field(default=None, init=False)
```

Comment:
I think it should be already enough to define the executor at the class level.
```suggestion
    executor: Final = None
```